File: python/function/log_callback.py
from ArducamEvkSDK import Camera, Param, get_error_name, LoggerLevel, CameraConfig


def log_callback(level, msg):
    """
    @type level: LoggerLevel
    @type msg: str
    """
    print("[{}] {}".format(level.name, msg))


def main():
    camera = Camera()
    param = Param()
    # open camera, return True if success, otherwise return False
    if not camera.open(param):
        # get the last error message
        raise Exception(
            "open camera error! {}".format(get_error_name(camera.last_error))
        )
    print("Disable Console Log")
    camera.set_message_callback(log_callback)  # set log callback
    camera.enable_console_log(False)  # disable console log
    camera.log_level = LoggerLevel.Trace  # set log level to info

    # set camera config
    camera_config = CameraConfig()
    camera_config.height = 1080  # set height (required)
    camera_config.width = 1920  # set width (required)
    camera.config = camera_config  # set config to camera

    # init camera
    camera.init()
    camera.close()


if __name__ == "__main__":
    main()

File: c/functions/log_callback.c
#include <arducam/arducam_evk_sdk.h>
#include <stdio.h>
#include <stdlib.h>

#include "options.h"

static const char *to_name(ArducamLoggerLevel level) {
    switch (level) {
    case trace:
        return "trace";
    case debug:
        return "debug";
    case info:
        return "info";
    case warn:
        return "warn";
    case err:
        return "err";
    case critical:
        return "critical";
    case off:
        return "off";
    }
}

void log_(ArducamLoggerLevel level, const char *msg, void *user_data) {
    // log callback
    printf("[%s] %s\n", to_name(level), msg);
}

void log_callback() {
    ArducamCameraHandle camera;
    ArducamCameraOpenParam param;
    ArducamDefaultParam(&param);  // init camera open parameters with the default values
    int ret =
        ArducamOpenCamera(&camera, &param);  // open camera, return Success if success, otherwise return error code
    if (ret != Success) {
        // get the last error message
        printf("open error! %s\n", ArducamErrorName(ret));
        exit(-1);
    }
    printf("Disable Console Log\n");
    ArducamDisableConsoleLog(camera);                    // disable console log
    ArducamSetLogLevel(camera, trace);                   // set log level to info
    ArducamRegisterMessageCallback(camera, log_, NULL);  // set log callback

    // set camera config
    ArducamCameraConfig config;
    config.height = 1080;                     // set height (camera, required)
    config.width = 1920;                      // set width (camera, required)
    ArducamSetCameraConfig(camera, &config);  // set config to camera

    // init camera
    ArducamInitCamera(camera);
    ArducamCloseCamera(camera);
}

int main(int argc, char **argv) {
    log_callback();
    return 0;
}
